src/main/java/net/clgd/ccemux/rendering/javafx/JFXTerminalFont.java

```diff
@@ -0,0 +1,67 @@
+package net.clgd.ccemux.rendering.javafx;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.net.URL;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import javafx.scene.image.Image;
+import javafx.scene.image.WritableImage;
+import javafx.scene.paint.Color;
+import lombok.*;
+import net.clgd.ccemux.rendering.TerminalFont;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class JFXTerminalFont extends TerminalFont {
+	@Getter(lazy = true)
+	private static final JFXTerminalFont bestFont = TerminalFont.getBest(JFXTerminalFont::new);
+
+	private final Image base;
+
+	private final Cache<Pair<Character, Color>, Image> charCache = CacheBuilder.newBuilder()
+			.expireAfterAccess(10, SECONDS).maximumSize(1000).initialCapacity(200).build();
+
+	public JFXTerminalFont(Image base) {
+		super(base.widthProperty().intValue(), base.heightProperty().intValue());
+		this.base = base;
+	}
+
+	public JFXTerminalFont(String url) {
+		this(new Image(url));
+	}
+
+	public JFXTerminalFont(URL url) {
+		this(url.toString());
+	}
+
+	public Image generateCharImage(char c, Color color) {
+		val coords = getCharCoords(c);
+		val out = new WritableImage(coords.width, coords.height);
+
+		val reader = base.getPixelReader();
+		val writer = out.getPixelWriter();
+
+		// TODO: probably faster to use a PixelFormat
+		for (int x = 0; x < coords.width; x++) {
+			for (int y = 0; y < coords.height; y++) {
+				Color oc = reader.getColor(coords.x + x, coords.y + y);
+
+				writer.setColor(x, y, Color.color(oc.getRed() * color.getRed(), oc.getGreen() * color.getGreen(),
+						oc.getBlue() * color.getBlue(), oc.getOpacity()));
```

Comment:
This again? :/ Is there no better way?

Reply:
There might be using `PixelFormat` to transform the colors as they're read and read an entire buffer at a time but I haven't had the time to test it yet